src/bio/bio_config.c

```diff
@@ -181,21 +186,26 @@ is_addr_in_allowlist(char *pci_addr, const struct spdk_pci_addr *allowlist,
 static int
 traddr_to_vmd(char *dst, const char *src)
 {
-	char		*traddr_tmp = NULL, *vmd_addr = NULL;
+	char            *traddr_tmp = NULL;
+	char            *vmd_addr   = NULL;
 	char		*ptr;
 	const char	 ch = ':';
 	char		 addr_split[3];
 	int		 position;
 	int		 iteration;
 	int		 n, rc = 0;
 	int		 vmd_addr_left_len;
+	int              len;
 
 	D_ALLOC(vmd_addr, SPDK_NVMF_TRADDR_MAX_LEN + 1);
 	if (vmd_addr == NULL)
 		return -DER_NOMEM;
 
 	strncat(vmd_addr, "0000:", SPDK_NVMF_TRADDR_MAX_LEN);
-	vmd_addr_left_len = SPDK_NVMF_TRADDR_MAX_LEN - strlen(vmd_addr);
+	len = strnlen(vmd_addr, SPDK_NVMF_TRADDR_MAX_LEN);
```

Comment:
[minor]  if SPDK_NVMF_TRADDR_MAX_LEN should be valid, it should be like:
len = strnlen(vm_addr, SPDK_NVMF_TRADDR_MAX_LEN + 1)
if (len == 0 || len ==  SPDK_NVMF_TRADDR_MAX_LEN + 1)
     return -DER_INVAL;

Reply:
why? the max refers to the maximum number of characters which is what strnlen returns (excluding the null terminator).

Reply:
yes I see, that correction is allowing for the case where `strlen(addr) == MAX`

Reply:
done